### PR TITLE
feat: add `ReadUser` function

### PR DESCRIPTION
**Pull request by barryib**
For https://github.com/Ouest-France/terraform-provider-ldap/pull/13

**Todo**

- [x] Add tests

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	if sAMAccountName != "" {
+		filter = fmt.Sprintf("%s(sAMAccountName=%s)", filter, sAMAccountName)
+	}
+
+	if upn != "" {
+		filter = fmt.Sprintf("%s(userPrincipalName=%s)", filter, upn)
+	}
+
+	filter = fmt.Sprintf("(&(objectCategory=person)(objectClass=user)%s)", filter)
```
Comment (pablo-ruth) [inline user.go:25]: At first read I was a bit confused by how the code build the filter, I suggest instead to use an array to store filter "conditions" and join them after. Something like that:
```
conditions := []string{"(objectCategory=person)", "(objectClass=user)"}

if name != "" {
	conditions = append(conditions, fmt.Sprintf("(name=%s)", name))
}

....

filter := fmt.Sprintf("(&%s)", strings.Join(conditions, ""))
```
What do you think?

**Review by pablo-ruth — CHANGES_REQUESTED** [review]

File: user.go (hunk 1 above)
Comment (barryib) [reply user.go:25]: Great. Your suggestion sounds good to me.

For the records, I'm not a go expert ^^

Comment (barryib) [reply user.go]: it's much better ^^

Comment (pablo-ruth) [reply user.go]: Great :)

**Review by pablo-ruth — APPROVED** [review]
LGTM

**Comment by pablo-ruth** [comment]
I tagged the version v0.4.0 so you can use it in Ouest-France/terraform-provider-ldap#13

**Pull request merged**